### PR TITLE
Add VASP setting for the dftd4 vdW functional and extend PBE_64 support.

### PR DESCRIPTION
**Pull request by hongyi-zhao**
This is an enhancement and re-implementation of https://github.com/materialsproject/pymatgen/pull/3899 and the PRs suggested to include in it.

Add VASP setting for the dftd4 vdW functional and extend `PBE_64` support:

- Add `IVDW: 13` for dftd4 in `vdW_parameters.yaml`
- Support `PBE_64` functional in `MPScanRelaxSet`, `MVLRelax52Set`, `MVLScanRelaxSet`, and `LobsterSet`
- Update documentation and error messages to include `PBE_64`
- Add warning for `LobsterSet` when using `PBE_52` and `PBE_64` POTCARs with `PBE_54` basis functions
- Update relevant test cases to accommodate `PBE_64` support

### CODE REVIEW

**Comment by hongyi-zhao** [comment]
I've installed the dev and optional dependencies per the guidance [here](https://pymatgen.org/contributing.html). There were no problems with my local pre-commit and pytest checks, So, I don't know why the PR failed to pass some of the tests.

File: src/pymatgen/io/vasp/sets.py [hunk 1]
```diff
@@ -2679,12 +2679,20 @@ class LobsterSet(VaspInputSet):
...
 
     CONFIG = MPRelaxSet.CONFIG
-    _valid_potcars: Sequence[str] | None = ("PBE_52", "PBE_54")
+    _valid_potcars: Sequence[str] | None = ("PBE_52", "PBE_54", "PBE_64")
 
     def __post_init__(self) -> None:
         super().__post_init__()
         warnings.warn("Make sure that all parameters are okay! This is a brand new implementation.")
 
+        if self.user_potcar_functional == "PBE_64":
```
Comment (JaGeo) [inline src/pymatgen/io/vasp/sets.py:2688]:
```suggestion
        if self.user_potcar_functional == "PBE_64":
```

Could you add PBE_52 here as well? Thanks!

**Comment by JaGeo** [comment]
And, @hongyi-zhao , thank you!

File: src/pymatgen/io/vasp/sets.py (hunk 1 above)
Comment (hongyi-zhao) [reply src/pymatgen/io/vasp/sets.py]: Done. Thank you for reminding me.

**Review by JaGeo — APPROVED** [review]
From my side, the changes are fine

**Review by janosh — APPROVED** [review]
looks good! thanks @hongyi-zhao 👍

File: src/pymatgen/io/vasp/sets.py (hunk 1 above)
Comment (janosh) [reply src/pymatgen/io/vasp/sets.py]: could you resolve the merge conflicts? `sets.py` is now split into submodules so it's a matter of re-applying your changes to the right file

**Comment by hongyi-zhao** [comment]
@shyuep I'll review and try to fix it.

File: src/pymatgen/io/vasp/sets.py (hunk 1 above)
Comment (hongyi-zhao) [reply src/pymatgen/io/vasp/sets.py]: This has been reverted by https://github.com/materialsproject/pymatgen/pull/3989.

**Comment by hongyi-zhao** [comment]
> I've installed the dev and optional dependencies per the guidance [here](https://pymatgen.org/contributing.html). There were no problems with my local pre-commit and pytest checks. So, I don't know why the PR failed to pass some of the tests.
> 
> My other PR, aka, #3955, encountered the same problem.

Now, all the above problems have been fixed.

**Pull request merged**